internal/db/sqlite/sql/migrations/folder/04-alter-blocks-tables.sql

```diff
@@ -0,0 +1,45 @@
+-- Copyright (C) 2025 The Syncthing Authors.
+--
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this file,
+-- You can obtain one at https://mozilla.org/MPL/2.0/.
+
+-- Copy blocks to new table with fewer indexes
+
+CREATE TABLE IF NOT EXISTS blocks_v4 (
```

Comment:
Shouldn't we fail if it exists - it should never? And to avoid failing if for some reason the table is there, do a `DROP blocks_v4 IF EXISTS` first - comes at the downside that if `blocks` doesn't exist either, we then have no more blocks. So probably just fail?

Reply:
It probably should be in the same transaction yes

Reply:
If it exists, it would be an error condition and the migration should fail IMHO.

+1 from me for dropping the check

Reply:
> the migration should fail

It should not, under any circumstances, really, because that will leave the user dead in the water with no way forward. I added an explicit drop instead. But, we do this in a transaction, so in principle it should not fail...